tests/BuildScriptGenerator.Tests/Node/NodeVersionCheckerTest.cs

```diff
@@ -0,0 +1,59 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System.Linq;
+using System.Collections.Generic;
+using Microsoft.Oryx.BuildScriptGenerator.Node;
+using Xunit;
+
+namespace Microsoft.Oryx.BuildScriptGenerator.Tests.Node
+{
+    public class NodeVersionCheckerTest
+    {
+        [Fact]
+        public void Checker_Warns_WhenOutdatedVersionUsed()
+        {
+            // Arrange
+            var checker = new NodeVersionChecker();
+
+            // Act
+            var messages = checker.CheckToolVersions(
+                new Dictionary<string, string> { { NodeConstants.NodeJsName, "1.0.0" } });
+
+            // Assert
+            Assert.Single(messages);
+            Assert.Contains("outdated version of Node.js was used", messages.First().Content);
+        }
+
+        [Fact]
```

Comment:
I would like this to be theory where we verify this for all the lts versions we support

Reply:
actually it'll be weird to implement since we don't have any seperation in the `TestValueGenerator` between "old" and "current" versions